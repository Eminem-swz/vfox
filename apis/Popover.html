<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Popover 气泡类型弹窗 | Vfox for Vue3</title>
    <meta name="generator" content="VuePress 1.8.2">
    
    <meta name="description" content="移动端 Vue 组件库">
    
    <link rel="preload" href="/vfox/assets/css/0.styles.7a77e6be.css" as="style"><link rel="preload" href="/vfox/assets/js/app.d94cf92d.js" as="script"><link rel="preload" href="/vfox/assets/js/2.9e7c9cf8.js" as="script"><link rel="preload" href="/vfox/assets/js/14.39f5d192.js" as="script"><link rel="prefetch" href="/vfox/assets/js/10.005c2f07.js"><link rel="prefetch" href="/vfox/assets/js/11.a0342a56.js"><link rel="prefetch" href="/vfox/assets/js/12.7a3dff27.js"><link rel="prefetch" href="/vfox/assets/js/13.79406c55.js"><link rel="prefetch" href="/vfox/assets/js/15.11a365eb.js"><link rel="prefetch" href="/vfox/assets/js/16.0f1d7e5a.js"><link rel="prefetch" href="/vfox/assets/js/17.1d88da16.js"><link rel="prefetch" href="/vfox/assets/js/18.fa0d6e12.js"><link rel="prefetch" href="/vfox/assets/js/19.69076c6e.js"><link rel="prefetch" href="/vfox/assets/js/20.f4248b7e.js"><link rel="prefetch" href="/vfox/assets/js/21.656ea408.js"><link rel="prefetch" href="/vfox/assets/js/22.0b0a326f.js"><link rel="prefetch" href="/vfox/assets/js/23.3b008b9a.js"><link rel="prefetch" href="/vfox/assets/js/24.0cd00f17.js"><link rel="prefetch" href="/vfox/assets/js/25.7f128a40.js"><link rel="prefetch" href="/vfox/assets/js/26.36d74912.js"><link rel="prefetch" href="/vfox/assets/js/27.201f1cbd.js"><link rel="prefetch" href="/vfox/assets/js/28.b4cbd27c.js"><link rel="prefetch" href="/vfox/assets/js/29.360868cc.js"><link rel="prefetch" href="/vfox/assets/js/3.828773b4.js"><link rel="prefetch" href="/vfox/assets/js/30.8e25e9ce.js"><link rel="prefetch" href="/vfox/assets/js/31.9eea5065.js"><link rel="prefetch" href="/vfox/assets/js/32.02ccbf62.js"><link rel="prefetch" href="/vfox/assets/js/33.c663930f.js"><link rel="prefetch" href="/vfox/assets/js/34.f552b726.js"><link rel="prefetch" href="/vfox/assets/js/35.420e15c4.js"><link rel="prefetch" href="/vfox/assets/js/36.899045cc.js"><link rel="prefetch" href="/vfox/assets/js/37.59d52f61.js"><link rel="prefetch" href="/vfox/assets/js/38.4cd203fc.js"><link rel="prefetch" href="/vfox/assets/js/39.a34326b1.js"><link rel="prefetch" href="/vfox/assets/js/4.a0de823d.js"><link rel="prefetch" href="/vfox/assets/js/40.f5f8d40b.js"><link rel="prefetch" href="/vfox/assets/js/41.32c9319a.js"><link rel="prefetch" href="/vfox/assets/js/42.80f5aa5e.js"><link rel="prefetch" href="/vfox/assets/js/43.d3a7fadd.js"><link rel="prefetch" href="/vfox/assets/js/44.be0c626c.js"><link rel="prefetch" href="/vfox/assets/js/45.c5e151d3.js"><link rel="prefetch" href="/vfox/assets/js/46.a59804fd.js"><link rel="prefetch" href="/vfox/assets/js/47.f67efbee.js"><link rel="prefetch" href="/vfox/assets/js/48.ee0614a0.js"><link rel="prefetch" href="/vfox/assets/js/49.7ec51df6.js"><link rel="prefetch" href="/vfox/assets/js/5.722fadb6.js"><link rel="prefetch" href="/vfox/assets/js/50.16910741.js"><link rel="prefetch" href="/vfox/assets/js/51.39fd4b95.js"><link rel="prefetch" href="/vfox/assets/js/52.5f60a0fe.js"><link rel="prefetch" href="/vfox/assets/js/53.24e450a8.js"><link rel="prefetch" href="/vfox/assets/js/54.2d06c015.js"><link rel="prefetch" href="/vfox/assets/js/55.f4c43543.js"><link rel="prefetch" href="/vfox/assets/js/56.b0f0aa5b.js"><link rel="prefetch" href="/vfox/assets/js/57.3aeca152.js"><link rel="prefetch" href="/vfox/assets/js/58.7c00827f.js"><link rel="prefetch" href="/vfox/assets/js/59.6de495dc.js"><link rel="prefetch" href="/vfox/assets/js/6.daa090d3.js"><link rel="prefetch" href="/vfox/assets/js/60.a75abd97.js"><link rel="prefetch" href="/vfox/assets/js/61.9dbf926d.js"><link rel="prefetch" href="/vfox/assets/js/62.724757a2.js"><link rel="prefetch" href="/vfox/assets/js/63.0d1bea98.js"><link rel="prefetch" href="/vfox/assets/js/64.eeb8f4d7.js"><link rel="prefetch" href="/vfox/assets/js/65.4ec922d9.js"><link rel="prefetch" href="/vfox/assets/js/66.8f77d484.js"><link rel="prefetch" href="/vfox/assets/js/67.a3170d35.js"><link rel="prefetch" href="/vfox/assets/js/68.c86961c8.js"><link rel="prefetch" href="/vfox/assets/js/69.1fb3d2d8.js"><link rel="prefetch" href="/vfox/assets/js/7.f522b1de.js"><link rel="prefetch" href="/vfox/assets/js/70.e912b407.js"><link rel="prefetch" href="/vfox/assets/js/71.0a8ff075.js"><link rel="prefetch" href="/vfox/assets/js/72.7806d0e1.js"><link rel="prefetch" href="/vfox/assets/js/73.fdb09df7.js"><link rel="prefetch" href="/vfox/assets/js/74.418e9dd4.js"><link rel="prefetch" href="/vfox/assets/js/75.aad3d4db.js"><link rel="prefetch" href="/vfox/assets/js/76.99be2e3d.js"><link rel="prefetch" href="/vfox/assets/js/77.a983e540.js"><link rel="prefetch" href="/vfox/assets/js/78.a9484581.js"><link rel="prefetch" href="/vfox/assets/js/79.9311664f.js"><link rel="prefetch" href="/vfox/assets/js/8.6dee2cb0.js"><link rel="prefetch" href="/vfox/assets/js/80.628ba263.js"><link rel="prefetch" href="/vfox/assets/js/81.ff48b0ec.js"><link rel="prefetch" href="/vfox/assets/js/82.4b16fff1.js"><link rel="prefetch" href="/vfox/assets/js/83.8b240d68.js"><link rel="prefetch" href="/vfox/assets/js/84.b72f2b7e.js"><link rel="prefetch" href="/vfox/assets/js/85.399677de.js"><link rel="prefetch" href="/vfox/assets/js/86.0c5c0b26.js"><link rel="prefetch" href="/vfox/assets/js/87.361b1ec2.js"><link rel="prefetch" href="/vfox/assets/js/88.f2342551.js"><link rel="prefetch" href="/vfox/assets/js/89.4d68079f.js"><link rel="prefetch" href="/vfox/assets/js/9.2e4fe4a6.js"><link rel="prefetch" href="/vfox/assets/js/90.2ef4436e.js"><link rel="prefetch" href="/vfox/assets/js/91.d9b5f188.js">
    <link rel="stylesheet" href="/vfox/assets/css/0.styles.7a77e6be.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/vfox/" class="home-link router-link-active"><img src="/vfox/assets/images/logo.png" alt="Vfox for Vue3" class="logo"> <span class="site-name can-hide">Vfox for Vue3</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/vfox/" class="nav-link">
  指南
</a></div><div class="nav-item"><a href="/vfox/design/" class="nav-link">
  设计
</a></div><div class="nav-item"><a href="/vfox/components/Button.html" class="nav-link">
  组件
</a></div><div class="nav-item"><a href="/vfox/apis/LocalStorage.html" class="nav-link">
  API
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="选择版本" class="dropdown-title"><span class="title">选择版本</span> <span class="arrow down"></span></button> <button type="button" aria-label="选择版本" class="mobile-dropdown-title"><span class="title">选择版本</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" class="nav-link external">
  1.x
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li><li class="dropdown-item"><!----> <a href="https://godxiaoji.github.io/vfox/" target="_blank" rel="noopener noreferrer" class="nav-link external">
  2.x
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li></ul></div></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/vfox/" class="nav-link">
  指南
</a></div><div class="nav-item"><a href="/vfox/design/" class="nav-link">
  设计
</a></div><div class="nav-item"><a href="/vfox/components/Button.html" class="nav-link">
  组件
</a></div><div class="nav-item"><a href="/vfox/apis/LocalStorage.html" class="nav-link">
  API
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="选择版本" class="dropdown-title"><span class="title">选择版本</span> <span class="arrow down"></span></button> <button type="button" aria-label="选择版本" class="mobile-dropdown-title"><span class="title">选择版本</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" class="nav-link external">
  1.x
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li><li class="dropdown-item"><!----> <a href="https://godxiaoji.github.io/vfox/" target="_blank" rel="noopener noreferrer" class="nav-link external">
  2.x
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li></ul></div></div> <!----></nav>  <ul class="sidebar-links"><li><a href="/vfox/" aria-current="page" class="sidebar-link">指南</a></li><li><a href="/vfox/design/" class="sidebar-link">设计</a></li><li><a href="/vfox/components/Button.html" class="sidebar-link">组件</a></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>API</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/vfox/apis/LocalStorage.html" class="sidebar-link">LocalStorage 本地缓存</a></li><li><a href="/vfox/apis/Notify.html" class="sidebar-link">Notify 通知类型弹窗</a></li><li><a href="/vfox/apis/Toast.html" class="sidebar-link">Toast 消息类型弹窗</a></li><li><a href="/vfox/apis/Scroll.html" class="sidebar-link">Scroll 滚动</a></li><li><a href="/vfox/apis/Preview.html" class="sidebar-link">Preview 预览类型弹窗</a></li><li><a href="/vfox/apis/Modal.html" class="sidebar-link">Modal 模态框弹窗</a></li><li><a href="/vfox/apis/Drawer.html" class="sidebar-link">Drawer 抽屉类型弹窗</a></li><li><a href="/vfox/apis/Popover.html" aria-current="page" class="active sidebar-link">Popover 气泡类型弹窗</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/vfox/apis/Popover.html#popover-showpopover-object" class="sidebar-link">Popover.showPopover(object)</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/vfox/apis/Popover.html#params" class="sidebar-link">Params</a></li><li class="sidebar-sub-header"><a href="/vfox/apis/Popover.html#usage" class="sidebar-link">Usage</a></li></ul></li><li class="sidebar-sub-header"><a href="/vfox/apis/Popover.html#popdialog-showpopdialog-object" class="sidebar-link">PopDialog.showPopDialog(object)</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/vfox/apis/Popover.html#params-2" class="sidebar-link">Params</a></li><li class="sidebar-sub-header"><a href="/vfox/apis/Popover.html#usage-2" class="sidebar-link">Usage</a></li></ul></li><li class="sidebar-sub-header"><a href="/vfox/apis/Popover.html#popmenu-showpopmenu-object" class="sidebar-link">PopMenu.showPopMenu(object)</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/vfox/apis/Popover.html#params-3" class="sidebar-link">Params</a></li><li class="sidebar-sub-header"><a href="/vfox/apis/Popover.html#usage-3" class="sidebar-link">Usage</a></li></ul></li></ul></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="popover-气泡类型弹窗"><a href="#popover-气泡类型弹窗" class="header-anchor">#</a> Popover 气泡类型弹窗</h1> <h2 id="popover-showpopover-object"><a href="#popover-showpopover-object" class="header-anchor">#</a> Popover.showPopover(object)</h2> <p>显示气泡。</p> <h3 id="params"><a href="#params" class="header-anchor">#</a> Params</h3> <p>Object object</p> <table><thead><tr><th>属性</th> <th>类型</th> <th>默认值</th> <th>必填</th> <th>说明</th></tr></thead> <tbody><tr><td>selector</td> <td>string/element</td> <td></td> <td>是</td> <td>从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到</td></tr> <tr><td>placement</td> <td>string</td> <td>'bottom'</td> <td>否</td> <td>展开位置，可选 'bottom', 'top', 'left', 'right'</td></tr> <tr><td>content</td> <td>string</td> <td></td> <td>是</td> <td>文本内容</td></tr> <tr><td>showMask</td> <td>boolean</td> <td>true</td> <td>否</td> <td>是否展示蒙层，如果设置不展示，气泡则是跟随 <code>selector</code> 对应的元素</td></tr> <tr><td>success</td> <td>function</td> <td></td> <td>否</td> <td>接口调用成功的回调函数</td></tr> <tr><td>fail</td> <td>function</td> <td></td> <td>否</td> <td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr> <tr><td>complete</td> <td>function</td> <td></td> <td>否</td> <td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table> <h3 id="usage"><a href="#usage" class="header-anchor">#</a> Usage</h3> <div class="language- extra-class"><pre class="language-text"><code>import { Popover } from 'vfox'

Popover.showPopover({
  selector: '#popoverTarget',
  content: '这是气泡内容',
  success: res =&gt; {
    console.log('success', res)
  }
})
</code></pre></div><h2 id="popdialog-showpopdialog-object"><a href="#popdialog-showpopdialog-object" class="header-anchor">#</a> PopDialog.showPopDialog(object)</h2> <p>显示气泡对话框。</p> <h3 id="params-2"><a href="#params-2" class="header-anchor">#</a> Params</h3> <p>Object object</p> <table><thead><tr><th>属性</th> <th>类型</th> <th>默认值</th> <th>必填</th> <th>说明</th></tr></thead> <tbody><tr><td>selector</td> <td>string/element</td> <td></td> <td>是</td> <td>从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到</td></tr> <tr><td>placement</td> <td>string</td> <td>'bottom'</td> <td>否</td> <td>展开位置，可选 'bottom', 'top', 'left', 'right'</td></tr> <tr><td>content</td> <td>string</td> <td></td> <td>是</td> <td>文本内容</td></tr> <tr><td>showCancel</td> <td>boolean</td> <td>true</td> <td>否</td> <td>是否显示取消按钮</td></tr> <tr><td>cancelText</td> <td>string</td> <td>'取消'</td> <td>否</td> <td>取消按钮的文本</td></tr> <tr><td>confirmText</td> <td>string</td> <td>'确定'</td> <td>否</td> <td>确定按钮的文本</td></tr> <tr><td>success</td> <td>function</td> <td></td> <td>否</td> <td>接口调用成功（在用户做出选择后，如取消，确定）的回调函数</td></tr> <tr><td>fail</td> <td>function</td> <td></td> <td>否</td> <td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr> <tr><td>complete</td> <td>function</td> <td></td> <td>否</td> <td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table> <h4 id="object-success-回调参数"><a href="#object-success-回调参数" class="header-anchor">#</a> object.success 回调参数</h4> <table><thead><tr><th>属性</th> <th>类型</th> <th>说明</th></tr></thead> <tbody><tr><td>cancel?</td> <td>boolean</td> <td>为 true 时，表示取消</td></tr> <tr><td>confirm?</td> <td>boolean</td> <td>为 true 时，表示点击了确定</td></tr></tbody></table> <h3 id="usage-2"><a href="#usage-2" class="header-anchor">#</a> Usage</h3> <div class="language- extra-class"><pre class="language-text"><code>import { PopDialog } from 'vfox'

PopDialog.showPopDialog({
  selector: '#popDialogTarget',
  content: this.content,
  success: (res) =&gt; {
    console.log('success', res)
  }
})
</code></pre></div><h2 id="popmenu-showpopmenu-object"><a href="#popmenu-showpopmenu-object" class="header-anchor">#</a> PopMenu.showPopMenu(object)</h2> <p>显示气泡菜单。</p> <h3 id="params-3"><a href="#params-3" class="header-anchor">#</a> Params</h3> <p>Object object</p> <table><thead><tr><th>属性</th> <th>类型</th> <th>默认值</th> <th>必填</th> <th>说明</th></tr></thead> <tbody><tr><td>selector</td> <td>string/element</td> <td></td> <td>是</td> <td>从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到</td></tr> <tr><td>placement</td> <td>string</td> <td>'bottom'</td> <td>否</td> <td>展开位置，可选 'bottom', 'top', 'left', 'right'</td></tr> <tr><td>options</td> <td>array</td> <td></td> <td>否</td> <td>选项列表</td></tr> <tr><td>success</td> <td>function</td> <td></td> <td>否</td> <td>接口调用成功（在用户做出选择后，如取消，选择选项）的回调函数</td></tr> <tr><td>fail</td> <td>function</td> <td></td> <td>否</td> <td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr> <tr><td>complete</td> <td>function</td> <td></td> <td>否</td> <td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table> <h4 id="options-的结构"><a href="#options-的结构" class="header-anchor">#</a> options 的结构</h4> <table><thead><tr><th>options[index] 值</th> <th>类型</th> <th>默认值</th> <th>必填</th> <th>说明</th></tr></thead> <tbody><tr><td>name</td> <td>string</td> <td></td> <td>是</td> <td>选项名</td></tr> <tr><td>disabled</td> <td>string</td> <td>false</td> <td>否</td> <td>是否禁用</td></tr> <tr><td>icon</td> <td>string</td> <td></td> <td>否</td> <td>图标，使用 <a href="/vfox/components/Icon.html">Icon</a> 组件</td></tr></tbody></table> <div class="language- extra-class"><pre class="language-text"><code>[
  {
    name: '选项1',
    disabled: false,
    icon: 'MenuOutlined'
  }
]
</code></pre></div><h4 id="object-success-回调参数-2"><a href="#object-success-回调参数-2" class="header-anchor">#</a> object.success 回调参数</h4> <table><thead><tr><th>属性</th> <th>类型</th> <th>说明</th></tr></thead> <tbody><tr><td>confirm?</td> <td>boolean</td> <td>为 true 时，表示用户点击了选项，此时返回 <code>detail</code></td></tr> <tr><td>cancel?</td> <td>boolean</td> <td>为 true 时，表示用户点击了取消</td></tr> <tr><td>detail?.item</td> <td>any</td> <td>options[index] 的副本</td></tr> <tr><td>detail?.index</td> <td>number</td> <td>索引</td></tr></tbody></table> <h3 id="usage-3"><a href="#usage-3" class="header-anchor">#</a> Usage</h3> <div class="language- extra-class"><pre class="language-text"><code>import { PopMenu } from 'vfox'

PopMenu.showPopMenu({
  selector: '#popMenuTarget',
  options: [{
    icon: 'HeartOutlined',
    name: '爱心'
  },
  {
    icon: 'StarOutlined',
    name: '星星'
  },
  {
    icon: 'CircleOutlined',
    name: '圈圈',
    disabled: true
  }],
  placement: 'top',
  success: res =&gt; {
    console.log('select', res)
  }
})
</code></pre></div></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/vfox/apis/Drawer.html" class="prev">
        Drawer 抽屉类型弹窗
      </a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/vfox/assets/js/app.d94cf92d.js" defer></script><script src="/vfox/assets/js/2.9e7c9cf8.js" defer></script><script src="/vfox/assets/js/14.39f5d192.js" defer></script>
  </body>
</html>
