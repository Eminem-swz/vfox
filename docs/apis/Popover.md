# Popover 气泡

## showPopover(object)

显示气泡。

### object

| 属性      | 类型                  | 默认值   | 必填 | 说明                                                                              |
| --------- | --------------------- | -------- | ---- | --------------------------------------------------------------------------------- |
| selector  | string \| HTMLElement |          | 是   | 从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到 |
| placement | string                | 'bottom' | 否   | 展开位置，可选 'bottom', 'top', 'left', 'right'                                   |
| content   | string                |          | 是   | 文本内容                                                                          |
| showMask  | boolean               | true     | 否   | 是否展示蒙层，如果设置不展示，气泡则是跟随 `selector` 对应的元素                  |
| success   | function              |          | 否   | 接口调用成功的回调函数                                                            |
| fail      | function              |          | 否   | 接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）          |
| complete  | function              |          | 否   | 弹窗关闭或调用失败的回调函数                                                      |

### Usage

具体调用方式可以参考[API 调用](../guide/import.md#api-调用)。

```JavaScript
showPopover({
  selector: '#popoverTarget',
  content: '这是气泡内容',
  success: res => {
    console.log('success', res)
  }
})
```
