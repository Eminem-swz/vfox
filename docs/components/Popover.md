# Popover 气泡

## Import

```JavaScript
import { Popover } from 'vfox'
```

具体的引入方式可以参考[引入组件](../guide/import.md)。

## Props

| 属性            | 类型                  | 默认值   | 必填 | 说明                                                                              |
| --------------- | --------------------- | -------- | ---- | --------------------------------------------------------------------------------- |
| v-model:visible | boolean               | false    | 否   | 是否显示                                                                          |
| selector        | string \| HTMLElement |          | 是   | 从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到 |
| placement       | PlacementType         | 'bottom' | 否   | 展开位置，可选 'bottom', 'top', 'left', 'right'                                   |
| show-mask       | boolean               | true     | 否   | 是否展示蒙层，如果设置不展示，气泡则是跟随 `selector` 对应的元素                  |
| content         | string                |          | 否   | 展示文本，如设置了 `slot`，则不使用该字段                                         |

## Events

| 事件                 | 描述           | 回调函数参数                                                         | TypeScript 函数           |
| -------------------- | -------------- | -------------------------------------------------------------------- | ------------------------- |
| visible-state-change | 展示隐藏时触发 | payload: { state: [VisibleState](./Popover.md#visiblestate-值说明) } | PopupOnVisibleStateChange |

### VisibleState 值说明

| 值     | 说明                 | 备注                                              |
| ------ | -------------------- | ------------------------------------------------- |
| show   | 展示时触发           |                                                   |
| shown  | 展示且动画结束后触发 |                                                   |
| hide   | 隐藏时触发           | 可能携带其他参数 cancel, maskClick, closeClick 等 |
| hidden | 隐藏且动画结束后触发 | 可能携带其他参数 cancel, maskClick, closeClick 等 |

## Slots

### 默认（#default）

```Vue
<fx-popover>
  自定义内容
</fx-popover>
```
