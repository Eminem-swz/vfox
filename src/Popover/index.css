.fx-popover_arrow {
  position: absolute;
  width: 10px;
  height: 10px;
  transform: rotate(45deg);
  background-color: #fff;
  opacity: 0;
  transition: transform 0.2s;
  z-index: 0;
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.1);
}

.fx-popover_inner {
  position: absolute;
  border-radius: 4px;
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.1);
  opacity: 0;
  transition: transform 0.2s;
  transform: translateZ(0);
}

.fx-popover_content {
  position: relative;
  z-index: 1;
  min-height: 40px;
  min-width: 40px;
  background-color: #fff;
  border-radius: 4px;
  color: #262626;
  transform: translateZ(0);
  overflow: hidden;
}

.fx-popover_text {
  font-size: 16px;
  line-height: 22px;
  padding: 16px;
  white-space: pre-line;
  word-break: break-all;
  word-wrap: break-word;
}

.fx-popover.visible .fx-popover_inner {
  opacity: 1;
}

.fx-popover.visible .fx-popover_arrow {
  opacity: 1;
}
