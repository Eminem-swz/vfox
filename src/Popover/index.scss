@import '../style/var.scss';

.#{$prefix}-popover {
  &_arrow {
    position: absolute;
    width: 10px;
    height: 10px;
    transform: rotate(45deg);
    background-color: #fff;
    opacity: 0;
    transition: transform 0.2s;
    z-index: 0;
    box-shadow: 0 0 8px rgba($color: $black-color, $alpha: 0.1);
  }

  &_inner {
    position: absolute;
    border-radius: 4px;
    box-shadow: 0 0 8px rgba($color: $black-color, $alpha: 0.1);
    opacity: 0;
    transition: transform 0.2s;
    transform: translateZ(0);
  }

  &_content {
    position: relative;
    z-index: 1;
    min-height: 40px;
    min-width: 40px;
    background-color: #fff;
    border-radius: 4px;
    color: $title-color;
    transform: translateZ(0);
    overflow: hidden;
  }

  &_text {
    font-size: 16px;
    line-height: 22px;
    padding: 16px;
    white-space: pre-line;
    word-break: break-all;
    word-wrap: break-word;
  }

  &.visible {
    .#{$prefix}-popover {
      &_inner {
        opacity: 1;
      }

      &_arrow {
        opacity: 1;
      }
    }
  }
}
