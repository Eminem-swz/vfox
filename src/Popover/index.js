import Popover from './Popover.vue'
import { showPopover } from '../apis/Popup'

Popover.install = function(app) {
  app.component(Popover.name, Popover)

  // 添加api
  app.config.globalProperties.$showPopover = showPopover
}

Popover.showPopover = showPopover

export default Popover
