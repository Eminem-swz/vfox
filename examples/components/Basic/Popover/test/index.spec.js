/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Popover from '@/Popover'

const selector = '#test'

describe('Popover', () => {
  test('should render prop set content correctly', () => {
    const wrapper = mount(Popover, {
      propsData: {
        selector
      },
      content: 'content'
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop placement="left" correctly', () => {
    const wrapper = mount(Popover, {
      propsData: {
        selector,
        placement: 'left'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop visible=true correctly', () => {
    const wrapper = mount(Popover, {
      propsData: {
        selector,
        visible: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
