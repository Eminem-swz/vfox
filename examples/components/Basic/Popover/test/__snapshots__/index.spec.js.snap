// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover should render prop placement="left" correctly 1`] = `
<div class="fx-popover fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-mask"></div>
  <div class="fx-popover_inner" style="left: 0px; top: 0px; max-width: -16px; max-height: -16px;"><i class="fx-popover_arrow" style="left: 200vw; top: 200vh;"></i>
    <div class="fx-popover_content">
      <div class="fx-popover_text"></div>
    </div>
  </div>
</div>
`;

exports[`Popover should render prop set content correctly 1`] = `
<div class="fx-popover fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-mask"></div>
  <div class="fx-popover_inner" style="left: 0px; top: 0px; max-width: -16px; max-height: -16px;"><i class="fx-popover_arrow" style="left: 200vw; top: 200vh;"></i>
    <div class="fx-popover_content">
      <div class="fx-popover_text"></div>
    </div>
  </div>
</div>
`;

exports[`Popover should render prop visible=true correctly 1`] = `
<div class="fx-popover fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-mask"></div>
  <div class="fx-popover_inner" style="left: 0px; top: 0px; max-width: -16px; max-height: -16px;"><i class="fx-popover_arrow" style="left: 200vw; top: 200vh;"></i>
    <div class="fx-popover_content">
      <div class="fx-popover_text"></div>
    </div>
  </div>
</div>
`;
